biofefi/pages/3_Feature_Importance.py

```diff
@@ -237,64 +238,90 @@ def pipeline(
 base_dir = biofefi_experiments_base_dir()
 
-if experiment_name:
-    st.session_state[ConfigStateKeys.ExperimentName] = experiment_name
-
-    data_choices = os.listdir(base_dir / experiment_name)
-    data_choices = filter(lambda x: x.endswith(".csv"), data_choices)
 
-    model_choices = os.listdir(ml_model_dir(base_dir / experiment_name))
-    model_choices = [x for x in model_choices if x.endswith(".pkl")]
+if experiment_name:
 
-    explain_all_models = st.toggle(
-        "Explain all models", key=ConfigStateKeys.ExplainAllModels
+    fi_options = find_previous_fi_results(
+        biofefi_experiments_base_dir() / experiment_name
     )
 
-    if explain_all_models:
-        st.session_state[ConfigStateKeys.ExplainModels] = model_choices
+    if fi_options:
+        st.warning("You have run feature importance in this experiment previously.")
+        st.checkbox(
+            "Would you like to rerun feature importance? This will overwrite the existing results.",
+            value=True,
+            key=ConfigStateKeys.RerunFI,
+        )
     else:
-        model_selector(model_choices)
-
-    if model_choices := st.session_state.get(ConfigStateKeys.ExplainModels):
-        fi_options_form()
-
-        if st.button("Run Feature Importance", type="primary"):
-            config = build_configuration()
-            # save FI options
-            fi_options_file = fi_options_path(base_dir / experiment_name)
-            save_options(fi_options_file, config[1])
-            # save Fuzzy options if configured
-            if config[0] is not None:
-                fuzzy_options_file = fuzzy_options_path(base_dir / experiment_name)
-                save_options(fi_options_file, config[0])
-
-            process = Process(target=pipeline, args=config, daemon=True)
-            process.start()
-            cancel_button = st.button(
-                "Cancel", on_click=cancel_pipeline, args=(process,)
-            )
-            with st.spinner(
-                "Feature Importance pipeline is running in the background. "
-                "Check the logs for progress."
-            ):
-                # wait for the process to finish or be cancelled
-                process.join()
-            try:
-                st.session_state[ConfigStateKeys.FILogBox] = get_logs(
-                    log_dir(base_dir / experiment_name) / "fi"
-                )
-                st.session_state[ConfigStateKeys.FuzzyLogBox] = get_logs(
-                    log_dir(base_dir / experiment_name) / "fuzzy"
-                )
-                log_box(
-                    box_title="Feature Importance Logs", key=ConfigStateKeys.FILogBox
+        st.session_state[ConfigStateKeys.RerunFI] = True
+
+    if st.session_state[ConfigStateKeys.RerunFI]:
+
+        st.session_state[ConfigStateKeys.ExperimentName] = experiment_name
+
+        data_choices = os.listdir(base_dir / experiment_name)
+        data_choices = filter(lambda x: x.endswith(".csv"), data_choices)
```

Comment:
The data file used for the experiment is on the execution options. It's loaded in the `pipeline` function. Also it looks like `data_choices` isn't actually being used for anything so you can remove it